onnxruntime/core/session/ep_api.cc

```diff
@@ -86,6 +90,63 @@ ORT_API(const char*, NodeComputeContext_NodeName, _In_ const OrtNodeComputeConte
   return compute_context->node_name;
 }
 
+ORT_API_STATUS_IMPL(EpDevice_AddAllocatorInfo, _In_ OrtEpDevice* ep_device,
+                    _In_ const OrtMemoryInfo* allocator_memory_info) {
+  const OrtDevice& info = allocator_memory_info->device;
+  if (info.MemType() == OrtDevice::MemType::DEFAULT) {
+    ep_device->device_memory_info = allocator_memory_info;
+  } else {
+    ep_device->host_accessible_memory_info = allocator_memory_info;
```

Comment:
nit: should we explicitly check for `OrtDevice::MemType::HOST_ACCESSIBLE` to make it more explicit and return an error status otherwise (in case a new enum value is added later).